meta/SelfEnergies.m

```diff
@@ -618,29 +618,49 @@ therefore not be accessed in the form Glu(gO2).
            Utils`StartProgressBar[Dynamic[k], Length[nPointFunctions]];
            For[k = 1, k <= Length[nPointFunctions], k++,
                Utils`UpdateProgressBar[k, Length[nPointFunctions]];
-               {p,d} = CreateNPointFunction[nPointFunctions[[k]], vertexFunctionNames];
-               prototypes = prototypes <> p;
-               defs = defs <> d;
-               {p,d} = CreateNPointFunctionMatrix[nPointFunctions[[k]]];
-               prototypes = prototypes <> p;
-               defs = defs <> d;
+               {prototype,def} = CreateNPointFunction[nPointFunctions[[k]], vertexFunctionNames];
+               prototypes = prototypes <> prototype;
+               defs = defs <> def;
+               {prototype,def} = CreateNPointFunctionMatrix[nPointFunctions[[k]]];
+               prototypes = prototypes <> prototype;
+               defs = defs <> def;
            ];
            (* create derivatives of Higgs boson self-energies w.r.t. p^2 *)
            If[ValueQ[SARAH`HiggsBoson],
-              derivatives = Cases[nPointFunctions, FSSelfEnergy[SARAH`HiggsBoson | SARAH`HiggsBoson[__], ___]] /. {
-                  FSSelfEnergy -> SelfEnergies`FSSelfEnergyDerivative,
-                  A0[__] -> 0,
-                  B0 -> SelfEnergies`DB0,
-                  G0 -> SelfEnergies`DG0
-              };
+              derivatives = Cases[nPointFunctions, FSSelfEnergy[SARAH`HiggsBoson | SARAH`HiggsBoson[__], ___]];
+
+	      Unprotect[D];
```

Comment:
I have a proposal to avoid the un-protection and protection of `D`: We could use an up-value definition here:
~~~.m
D[SelfEnergies`FSSelfEnergy[particle_, expr_], mom2_] ^:= SelfEnergies`FSSelfEnergyDerivative[particle, D[expr, mom2]];
~~~
In this way Mathematica would not complain about `D` being protected. We could then omit the `Unprotect[D];` and `Protect[D];` statements.

Reply:
I have the feeling that the statements
~~~.m
D[...] ^:= ...
Derivative[...][...][...] := ...
~~~
would be better placed outside of the function, i.e. in the scope of the `SelfEnergies.m` module. In this way we could call the function multiple times. The alternative would be to use a replacement rule instead of definitions.